### PR TITLE
Add "replace" query param to Grok bulk import REST API endpoint.

### LINKED ISSUE #937: multiple grok patterns with the same name can be created
Running 1.0.0-8.rc.1from the Debian packages.

It is possible to create multiple grok patterns with the same name, both via the "create pattern" dialog and by importing from file (both by importing the same file twice, and by having multple patterns with the same name in a single file).

What should happen:

* When using the "create pattern" dialog, it should be impossible to save a pattern with an already-existing name.

* When importing a file, some sort of conflict resolution would be needed. This could be on a per-file level: if the newly imported file contains names that are already present, the user should have the choice to:
  * cancel processing of that file
  * overwrite existing patterns

  This would cover my usecase of keeping grok patterns in files, editing them there and importing them. Other usecases may be better served by a conflict resolution on a per-pattern basis.

### PR DESCRIPTION
**Pull request by bernd**
Existing Grok patterns will be removed before adding the new ones if set
to true.

Fixes #937
Refs Graylog2/graylog2-web-interface#1150

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/GrokResource.java [hunk 1]
```diff
@@ -104,6 +107,9 @@ public Response bulkUpdatePatterns(@ApiParam(name = "patterns", required = true)
                 throw new ValidationException("Invalid pattern " + pattern + ". Did not save any patterns.");
             }
         }
+        if (replace) {
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/GrokResource.java]: This logic should be moved into `GrokPatternService`.

Comment (bernd) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/GrokResource.java]: Then we have to move the pattern creation as well. Any suggestion for method names?

File: graylog2-server/src/main/java/org/graylog2/grok/GrokPatternServiceImpl.java [hunk 2]
```diff
@@ -98,4 +98,9 @@ public boolean validate(GrokPattern pattern) {
     public int delete(String patternId) {
         return dbCollection.removeById(new ObjectId(patternId)).getN();
     }
+
+    @Override
+    public void deleteAll() {
+        dbCollection.drop();
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/grok/GrokPatternServiceImpl.java]: While this *currently* has the same effect, I would replace this with a proper call to [`JacksonDBCollection#remove(com.mongodb.DBObject)`](http://mongojack.org/apidocs/org/mongojack/JacksonDBCollection.html#remove(com.mongodb.DBObject)).

File: graylog2-server/src/main/java/org/graylog2/rest/resources/system/GrokResource.java [hunk 3]
```diff
@@ -96,7 +97,9 @@ public Response createPattern(@ApiParam(name = "pattern", required = true)
     @PUT
     @Timed
     @ApiOperation("Add a list of new patterns")
-    public Response bulkUpdatePatterns(@ApiParam(name = "patterns", required = true) @NotNull GrokPatternList patternList) throws ValidationException {
+    public Response bulkUpdatePatterns(@ApiParam(name = "patterns", required = true) @NotNull GrokPatternList patternList,
+                                       @ApiParam(name = "replace", value = "Replace all patterns with the new ones.")
+                                       @QueryParam("replace") boolean replace) throws ValidationException {
```
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/rest/resources/system/GrokResource.java]: Maybe add a default value for `replace` (just to be sure) with [`@DefaultValue`](https://jersey.java.net/apidocs/2.16/jersey/javax/ws/rs/DefaultValue.html).

**Pull request merged**